docs/architecture/adr-006-subspace-module.md

```diff
@@ -0,0 +1,157 @@
+
+### Types 
+Subspaces must always have an account that is elected as the _owner_ and should act as the final decision maker and accountable user. Additionally, each subspace can have the following data:
+
+* a _human-readable name_ and a _description_, in order to allow users to easily identify the scope of such subspace;
+* a set of _connections_ with external applications, allowing users to verify the validity of such subspace and avoid any possible fakes;
+* an _ACL_ (access-control list) defining the permissions that different kind of users have inside the subspace itself (e.g. administrators, moderators, etc.).
+
+#### Subspace
+
+```go
+// Subspace contains all the data related to a Desmos subspace
+type Subspace struct {
+  // Unique identifier of this subspace
+  ID uint64
+
+  // Human-readable name of the subspace
+  Name string
+  
+  // Optional description of this subspace
+  Description string
+  
+  // Address of the user that owns the subspace
+  Owner  string
+
+  // Address of the subspace creator
+  Creator string
+  
+  // Represents the account that is associated with the subspace and
+  // should be used to connect external applications to verify this subspace
+  Treasury string
+
+  // the creation time of the subspace
+  CreationTime time.Time
+}
+```
+
+#### ACL
+In order to easily implement an ACL, we will use a simple set of keys made as follows: 
+```
+ACLPrefix + User Address -> ACL Value
+```
+
+The `ACL Value` will be a simple binary value allowing us to perform bitwise operations to combine the following different permissions: 
+
+```go
+const (
+  // Identifies users that can create content inside the subspace
+  PermissionWrite           = 0b000001
+  
+  // Allows users to moderate contents of other users (e.g. deleting it) 
+  PermissionModerateContent = 0b000010
+  
+  // Allows to add a link for this subspace
+  PermissionAddLink         = 0b000100
+  
+  // Allows to change the information of the subspace
+  PermissionChangeInfo      = 0b001000
+  
+  // Allows to set other users' permissions (except PermissionSetPermissions)
+  PermissionSetPermissions  = 0b010000
+)
+```
+
+> **Note**:  
+> Only the `Owner` account will be able to grant other users the `PermissionSetPermissions`
+
+Using this kind of permissions will allow us to easily set permissions and check whether a user has a permission or not: 
+```go
+userPermissions := PermissionWrite | PermissionAddLink | PermissionChangeInfo
+
+canWrite := (userPermissions & PermissionWrite) == PermissionWrite  // True
+canModerateContent := (userPermissions & PermissionModerateContent) == PermissionModerateContent // False
+```
+
+### `Msg` Service
+We will allow the following operations to be performed:
+
+* delete contents that do not respect the ToS;
+* ban users that do not respect the ToS.
+
```

Comment:
```suggestion
* create a subspace
* edit a subspace
* set user permissions
```

Reply:
And I think we should allow the possibility to transfer the subspace ownership too

Reply:
>  I think we should allow the possibility to transfer the subspace ownership too
@bragaz This should be already handled by the `MsgEditSubspace` that allows to change the treasury address